src/main/scala/pl/project13/scala/akka/raft/Candidate.scala

```diff
@@ -32,28 +32,55 @@ private[raft] trait Candidate {
         stay() using includingThisVote.withVoteFor(m.currentTerm, m.clusterSelf)
       }
 
-    case Event(msg: RequestVote, m: ElectionMeta) if m.canVoteIn(msg.term) =>
-      sender ! VoteCandidate(m.currentTerm)
-      stay() using m.withVoteFor(msg.term, candidate())
+		case Event(msg: RequestVote, m: ElectionMeta) if msg.term < m.currentTerm =>
+			log.info("Rejecting RequestVote msg by {} in {}. Received stale {}.", candidate, m.currentTerm, msg.term)
+			candidate ! DeclineCandidate(m.currentTerm)
+			stay()
+
+		case Event(msg: RequestVote, m: ElectionMeta) if msg.term > m.currentTerm =>
+			log.info("Received newer {}. Current term is {}. Revert to follower state.", msg.term, m.currentTerm)
```

Comment:
perhaps you should forward the RequestVote message instead of dropping it?

e.g.

```
m.clusterSelf forward msg
goto(Follower) using m.forFollower(msg.term)
```